src/main/java/UI/FilterMenu.java

```diff
@@ -1,55 +1,122 @@
-                null,
-                "Which area are you interested in?",
-                "Choose Filter",
-                JOptionPane.QUESTION_MESSAGE,
-                null,
-                options,
-                options[0]);
-    }
+        JButton jButton = new JButton("Done");
+        jButton.setBounds(100, 100, 90, 20);
 
-    public void setType(FilterType type) {
-        this.type = type;
-    }
+        JLabel jLabel = new JLabel();
+        jLabel.setBounds(90, 100, 400, 100);
 
-    public void applyFilter(){
-        if (Objects.equals(getFilter, options[0])){
-            setType(new SexFilterMaleType());
-        } if (Objects.equals(getFilter, options[1])){
-            setType(new SexFilterFemaleType());
-        }else {
-            setType(new LocationFilterType());
+        add(jButton);
+        add(jComboBox);
+        add(jLabel);
+
+        // create 5 result buttons and store them in a list.
+        ArrayList<JButton> userList = new ArrayList<>();
+        JButton user1 = new JButton("");
+        JButton user2 = new JButton("");
+        JButton user3 = new JButton("");
+        JButton user4 = new JButton("");
+        JButton user5 = new JButton("");
+
+        userList.add(user1);
+        userList.add(user2);
+        userList.add(user3);
+        userList.add(user4);
+        userList.add(user5);
+
+        // add the buttons to the frame, and set their location.
+        add(user1);
+        add(user2);
+        add(user3);
+        add(user4);
+        add(user5);
+
+        user1.setBounds(100,200,350,30);
+        user2.setBounds(200,200,350,30);
+        user3.setBounds(100,300,350,30);
+        user4.setBounds(200,300,350,30);
+        user5.setBounds(150,400,350,30);
+
+
+        //set the buttons to invisible and disabled
+        for (JButton users : userList){
+            users.setVisible(false);
+            users.setEnabled(false);
         }
-        filterController.setFilter(this.type);
-        filterController.performFilter();
-    }
 
-    public void setResult (UserAccount[] results){ this.results = results;}
+        jButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                String selectedFilter = "You selected " + jComboBox.getItemAt(jComboBox.getSelectedIndex());
+                jLabel.setText(selectedFilter);
+                // apply user's selection here
+                applyFilter(jComboBox.getItemAt(jComboBox.getSelectedIndex()));
+                //presenter the result and call the view profile UI when button is clicked
+                for (int i = 0; i <= results.length - 1; i++){
+                    if (i <= 4){
+                        JButton currButton = userList.get(i);
+                        UserAccount targetUser = results[i];
+                        currButton.setText(targetUser.getUsername());
+                        currButton.setVisible(true);
```

Comment:
At a glance, I am concerned that this code looks a bit smelly since the indentation's really high. You might want to fix that up in a later submission.